From a0cdb50d96799bc205283b6572e100a515e2870c Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Sat, 5 Dec 2009 19:33:38 +0800
Subject: [PATCH] ENGR00119023-1 MX51: Build in SPI NOR flash driver

Add CONFIG_MTD_MXC_DATAFLASH to build in SPI NOR
flash driver

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 arch/arm/configs/imx51_defconfig  |    3 ++-
 arch/arm/mach-mx51/mx51_babbage.c |    3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/configs/imx51_defconfig b/arch/arm/configs/imx51_defconfig
index 70fda11..feb7efe 100644
--- a/arch/arm/configs/imx51_defconfig
+++ b/arch/arm/configs/imx51_defconfig
@@ -1,7 +1,7 @@
 #
 # Automatically generated make config: don't edit
 # Linux kernel version: 2.6.31
-# Fri Nov 20 17:50:26 2009
+# Sat Dec  5 00:11:48 2009
 #
 CONFIG_ARM=y
 CONFIG_SYS_SUPPORTS_APM_EMULATION=y
@@ -529,6 +529,7 @@ CONFIG_MTD_CFI_I2=y
 # Self-contained MTD device drivers
 #
 # CONFIG_MTD_DATAFLASH is not set
+CONFIG_MTD_MXC_DATAFLASH=y
 # CONFIG_MTD_M25P80 is not set
 # CONFIG_MTD_SLRAM is not set
 # CONFIG_MTD_PHRAM is not set
diff --git a/arch/arm/mach-mx51/mx51_babbage.c b/arch/arm/mach-mx51/mx51_babbage.c
index 11955f8..c94dbf4 100644
--- a/arch/arm/mach-mx51/mx51_babbage.c
+++ b/arch/arm/mach-mx51/mx51_babbage.c
@@ -644,7 +644,8 @@ static struct spi_board_info mxc_spi_nor_device[] __initdata = {
 };
 
 static struct spi_board_info mxc_dataflash_device[] __initdata = {
-#if defined(CONFIG_MTD) || defined(CONFIG_MTD_MODULE)
+#if defined(CONFIG_MTD_MXC_DATAFLASH) \
+	|| defined(CONFIG_MTD_MXC_DATAFLASH_MODULE)
 	{
 	 .modalias = "mxc_dataflash",
 	 .max_speed_hz = 25000000,	/* max spi clock (SCK) speed in HZ */
-- 
1.5.4.4

